### PR TITLE
Move HoldToQuit button from HUDOverlay to PauseContainer

### LINKED ISSUE #3762: Pausing impossible while score overlay is hidden
Toggling the overlay on with Shift+Tab allows pausing as expected.

### PR DESCRIPTION
**Pull request by UselessToucan**
Closes #3762

### CODE REVIEW

**Comment by UselessToucan** [comment]
The only reason is that the button is related to pause. I'm ok with moving it to the `Player`.

**Comment by peppy** [comment]
Which would make sense if the action was assigned from within PauseContainer, but not so sure in its current state. Also the button sometimes exits instead of pauses.

File: osu.Game/Screens/Play/PauseContainer.cs [hunk 1]
```diff
@@ -72,8 +77,17 @@ public PauseContainer(FramedClock framedClock, DecoupleableInterpolatingFramedCl
                 OnRetry = () => OnRetry(),
                 OnQuit = () => OnQuit(),
             });
+
+            AddInternal(HoldToQuit = createHoldForMenuButton());
         }
 
+        private HoldForMenuButton createHoldForMenuButton() => new HoldForMenuButton
```
Comment (peppy) [inline osu.Game/Screens/Play/PauseContainer.cs]: this function is pointless, just inline it.

**Comment by peppy** [comment]
Do you understand what you're doing here? As in taking a logically sound piece of code and obliterating it? Hopefully you can see that.

**Pull request closed without merging**